substratevm/mx.substratevm/mx_substratevm.py

```diff
@@ -1193,6 +1193,19 @@ def mx_post_parse_cmd_line(opts):
     for dist in suite.dists:
         if dist.isJARDistribution():
             dist.set_archiveparticipant(GraalArchiveParticipant(dist, isTest=dist.name.endswith('_TEST')))
+    # Compilation of module-info.java classes need upgrade-module path arguments to
+    # when javac is invoked. This is in particular needed when the base JDK includes no
+    # JMODs.
+    if opts.no_jlinking:
+        all_jar_dists = set()
+        for p in suite.projects_recursive():
+            if p.isJavaProject():
+                jd = p.get_declaring_module_distribution()
+                if jd:
+                    all_jar_dists.add(jd)
+        for d in all_jar_dists:
+            d.set_extra_javac_processor(NoJlinkGraalJavacArgsProcessor(d, "jdk.graal.compiler"))
```

Comment:
You would need this after applying https://github.com/graalvm/mx/pull/287#discussion_r1858832058:
```suggestion
            d.set_extra_javac_processor(NoJlinkGraalJavacArgsProcessor(d, "jdk.graal.compiler").__process__)
```

Reply:
Thanks